src/backend/distributed/planner/recursive_planning.c

```diff
@@ -482,13 +483,15 @@ ContainsReferencesToOuterQueryWalker(Node *node, VarLevelsUpWalkerContext *conte
  *   read_intermediate_result('<planId>_<subPlanId>', '<copy format'>)
  *   AS res (<column definition list>);
  *
- * The target list and column definition list are derived from the given subquery.
+ * The target list and column definition list are derived from the given subquery
+ * and the cte columm name alias list.
  *
  * If any of the types in the target list cannot be used in the binary copy format,
  * then the copy format 'text' is used, otherwise 'binary' is used.
  */
 static Query *
-BuildSubPlanResultQuery(Query *subquery, uint64 planId, uint32 subPlanId)
+BuildSubPlanResultQuery(Query *subquery, CommonTableExpr *cte, uint64 planId,
```

Comment:
We're going to call this function for subqueries as well. So we'd better not to pass cte here?

Reply:
Maybe pass `List *aliascolnames `?

Reply:
makes sense